modules/dnn_superres/test/test_dnn_superres.cpp

```diff
@@ -0,0 +1,284 @@
+
+                void runOneModel(std::string algorithm, int scale, std::string model_filename);
+        };
+
+        void CV_DnnSuperResSingleOutputTest::runOneModel(std::string algorithm, int scale, std::string model_filename)
+        {
+            std::string path = std::string(ts->get_data_path()) + DNN_SUPERRES_DIR + "/" + IMAGE_FILENAME;
+
+            Mat img = imread(path);
+            if (img.empty())
+            {
+                ts->printf(cvtest::TS::LOG, "Test image not found!\n");
+                ts->set_failed_test_info(cvtest::TS::FAIL_INVALID_TEST_DATA);
+                return;
+            }
+
+            std::string pb_path = std::string(ts->get_data_path()) + DNN_SUPERRES_DIR + "/" + model_filename;
+
+            this->dnn_sr->readModel(pb_path);
+
+            this->dnn_sr->setModel(algorithm, scale);
+
+            if (this->dnn_sr->getScale() != scale)
+            {
+                ts->printf(cvtest::TS::LOG,
+                           "Scale factor could not be set for scale algorithm %s and scale factor %d!\n",
+                           algorithm.c_str(), scale);
+                ts->set_failed_test_info(cvtest::TS::FAIL_BAD_ACCURACY);
+                return;
+            }
+
+            if (this->dnn_sr->getAlgorithm() != algorithm)
+            {
+                ts->printf(cvtest::TS::LOG, "Algorithm could not be set for scale algorithm %s and scale factor %d!\n",
+                           algorithm.c_str(), scale);
+                ts->set_failed_test_info(cvtest::TS::FAIL_BAD_ACCURACY);
+                return;
+            }
+
+            Mat img_new;
+            this->dnn_sr->upsample(img, img_new);
+
+            if (img_new.empty())
+            {
+                ts->printf(cvtest::TS::LOG,
+                           "Could not perform upsampling for scale algorithm %s and scale factor %d!\n",
+                           algorithm.c_str(), scale);
+                ts->set_failed_test_info(cvtest::TS::FAIL_BAD_ACCURACY);
+                return;
+            }
+
+            int new_cols = img.cols * scale;
+            int new_rows = img.rows * scale;
+            if (img_new.cols != new_cols || img_new.rows != new_rows)
+            {
+                ts->printf(cvtest::TS::LOG, "Dimensions are not correct for scale algorithm %s and scale factor %d!\n",
+                           algorithm.c_str(), scale);
+                ts->set_failed_test_info(cvtest::TS::FAIL_BAD_ACCURACY);
+                return;
+            }
+        }
+
+        CV_DnnSuperResSingleOutputTest::CV_DnnSuperResSingleOutputTest()
+        {
+            dnn_sr = makePtr<DnnSuperResImpl>();
+        }
+
+        void CV_DnnSuperResSingleOutputTest::run(int)
+        {
+            //x2
+            runOneModel("espcn", 2, "ESPCN_x2.pb");
+        }
+
+        TEST(CV_DnnSuperResSingleOutputTest, accuracy)
+        {
+            CV_DnnSuperResSingleOutputTest test;
+            test.safe_run();
```

Comment:
We don't need testing class here and below (it is from legacy testing system).
Just call `runOneModel(...)` directly (or move its content into test):
```
TEST(DnnSuperResSingleOutputTest, accuracy)
{
    cv::Ptr<DnnSuperResImpl> dnn_sr = makePtr<DnnSuperResImpl>();
    ... all your test code from runOneModel() ...
}
```

Reply:
I updated the test codes.